tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_ConservativeFromPrimitive.cpp

```diff
@@ -0,0 +1,129 @@
+                                                used_for_size);
+  const auto spatial_velocity_squared =
+      make_with_random_values<Scalar<DataType>>(generator, distribution,
+                                                used_for_size);
+  const auto lorentz_factor = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto specific_enthalpy = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto pressure = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto sqrt_det_spatial_metric =
+      make_with_random_values<Scalar<DataType>>(generator, distribution,
+                                                used_for_size);
+  const auto spatial_velocity_oneform =
+      make_with_random_values<tnsr::i<DataType, Dim>>(generator, distribution,
+                                                      used_for_size);
+
+  auto tilde_d = make_with_value<Scalar<DataType>>(used_for_size, 0.);
+  auto tilde_tau = make_with_value<Scalar<DataType>>(used_for_size, 0.);
+  auto tilde_s = make_with_value<tnsr::i<DataType, Dim>>(used_for_size, 0.);
+
+  RelativisticEuler::Valencia::conservative_from_primitive(
+      make_not_null(&tilde_d), make_not_null(&tilde_tau),
+      make_not_null(&tilde_s), rest_mass_density, specific_internal_energy,
+      spatial_velocity_oneform, spatial_velocity_squared, lorentz_factor,
+      specific_enthalpy, pressure, sqrt_det_spatial_metric);
+  CHECK_ITERABLE_APPROX(expected_tilde_d(rest_mass_density, lorentz_factor,
+                                         sqrt_det_spatial_metric),
+                        tilde_d);
+  CHECK_ITERABLE_APPROX(
+      expected_tilde_tau(rest_mass_density, specific_internal_energy,
+                         spatial_velocity_squared, lorentz_factor, pressure,
+                         sqrt_det_spatial_metric),
+      tilde_tau);
+  CHECK_ITERABLE_APPROX(
+      expected_tilde_s(rest_mass_density, spatial_velocity_oneform,
+                       lorentz_factor, specific_enthalpy,
+                       sqrt_det_spatial_metric),
+      tilde_s);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.RelativisticEuler.Valencia.ConservativeFromPrimitive",
+                  "[Unit][RelativisticEuler]") {
+  std::random_device r;
+  const auto seed = r();
+  std::mt19937 generator(seed);
+  INFO("seed = " << seed);
+  std::uniform_real_distribution<> distribution(0.0, 1.0);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_distribution = make_not_null(&distribution);
+
+  const double d = std::numeric_limits<double>::signaling_NaN();
+  test_conservative_to_primitive<1>(nn_generator, nn_distribution, d);
+  test_conservative_to_primitive<2>(nn_generator, nn_distribution, d);
+  test_conservative_to_primitive<3>(nn_generator, nn_distribution, d);
+
+  const DataVector dv(5);
```

Comment:
This comment is slightly out of scope, but should/do we have a signaling_NaN for DataVectors?

Reply:
they are filled with them if you only specify a size